### PR TITLE
[COST-3726] Updates for cluster capacity & fix node capacity edge cases

### PR DESCRIPTION
**Pull request by myersCody**
## Jira Ticket

[COST-3726](https://issues.redhat.com/browse/COST-3726)

## Description

This change will fix edge cases related to the node capacity changes:
1. 500s with multiple OCP accounts and one of the accounts has partial data. 
   - It was causing us to try to do a max of an integer and a null causing panda to fail resulting in a 500. 
   - The [changes I made here](https://github.com/project-koku/koku/pull/4427/files#diff-7856c4ba653f802f9901aa185f9dfcd3754d01fe20a6fa824b6443d31b88d48aR1163-R1166) fixed the panda logic.
2. 500s related to `group_by[tag:app]=*`
   - The other's that gets created for tag grouping was causing problems in my logic for calculating unused request & capacity.
   - The other's row had a usage value it just also happened to be none
   - The [changes I made here](https://github.com/project-koku/koku/pull/4427/files#diff-5e7ab40825448d63b40fdfb5514b41aa9ce853cc4ba741c55d78ce1854905892R91) fixed that.  
3. While hardening I noticed that that the monthly resolution capacity was off. We should be summing the daily capacity of each node to generate a monthly capacity. I [updated the logic](https://github.com/project-koku/koku/pull/4427/files#diff-5e7ab40825448d63b40fdfb5514b41aa9ce853cc4ba741c55d78ce1854905892R334-R339) here to address that. 
4. Include instance count  & units on the cluster capacity returns. 
5. Add unused pieces & instance count to the meta total

## Testing

Reproducer for the 500s: Run this on main and compare to this branch
```
@pytest.mark.cost_smoke
def test_api_reproduce_api500(
    application, cost_ocp_on_gcp_sources_static_last_month, cost_ocp_on_gcp_source_markup
):
    """Test to ingest ocp multi sources

    metadata:
    requirements: cost_onboarding, cost_provider_management, cost_ocp_support
    importance: medium
    """
    breakpoint()
    assert cost_ocp_on_gcp_sources_static_last_month
``` 
   

100% Smoke Test Pass: https://ci.ext.devshift.net/job/project-koku-koku-pr-check/9986/

I did some test driven development for this, the smoke test that I wrote to test this can be located in [this branch](https://gitlab.cee.redhat.com/insights-qe/iqe-cost-management-plugin/-/tree/node_capacity_playground)
(`node_capacity_playground`).
## Notes

...

### CODE REVIEW

File: koku/api/report/ocp/query_handler.py [hunk 1]
```diff
@@ -337,19 +345,16 @@ def _generate_capacity_subsets(self, annotations, level_key, group_list):
...
-                _calculate_unused(row)
-            elif self.resolution == "daily":
-                row[_capacity.key] = _capacity.resolution_total.get(row.get("date"), Decimal(0))
-            elif self.resolution == "monthly" and not self.parameters.get("start_date"):
-                row[_capacity.key] = _capacity.total
-            else:
-                row_date = datetime.datetime.strptime(row.get("date"), "%Y-%m").month
-                row[_capacity.key] = _capacity.resolution_total.get(row_date, Decimal(0))
-        return query_data, {_capacity.key: _capacity.total}
+    # def _get_node_capacity(self, query_data, _capacity):
```
Comment (myersCody) [inline koku/api/report/ocp/query_handler.py:348]: Gonna delete this.

File: koku/api/report/ocp/query_handler.py [hunk 2]
```diff
@@ -305,78 +325,78 @@ def execute_query(self):  # noqa: C901
...
-                row_date = datetime.datetime.strptime(row.get("date"), "%Y-%m").month
-                row[_capacity.key] = _capacity.resolution_total.get(row_date, Decimal(0))
-        return query_data, {_capacity.key: _capacity.total}
-
     def _get_cluster_capacity(self, query_data, _capacity):
         """Calculate the cluster capacity."""
         for row in query_data:
-            cluster_list = row.get("clusters")
+            row_date = row.get("date")
+            # I am not convinced we need this:
```
Comment (lcouzens) [inline koku/api/report/ocp/query_handler.py:378]: I am not convinced we need this comment :smiling_imp:

Comment (myersCody) [reply koku/api/report/ocp/query_handler.py:378]: 💀 lol 🙇

Comment (myersCody) [reply koku/api/report/ocp/query_handler.py:378]:
```suggestion
```

**Review by lcouzens — APPROVED** [review]

**Pull request merged**